### PR TITLE
Fix/use generic test naming style instead of schema test

### PR DESCRIPTION
**Pull request by b-moynihan**
This is a:
- [x] bug fix PR with no breaking changes — please ensure the base branch is `main`
- [ ] new functionality — please ensure the base branch is the latest `dev/` branch
- [ ] a breaking change — please ensure the base branch is the latest `dev/` branch

## Description & motivation
This PR aims to fix https://github.com/dbt-labs/dbt-utils/issues/485

## Checklist
- [x] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [x] BigQuery
    - [x] Postgres
    - [x] Redshift
    - [x] Snowflake
- [x] I followed guidelines to ensure that my changes will work on "non-core" adapters by:
    - [x] dispatching any new macro(s) so non-core adapters can also use them (e.g. [the `star()` source](https://github.com/fishtown-analytics/dbt-utils/blob/master/macros/sql/star.sql))
    - [x] using the `limit_zero()` macro in place of the literal string: `limit 0`
    - [x] using `dbt_utils.type_*` macros instead of explicit datatypes (e.g. `dbt_utils.type_timestamp()` instead of `TIMESTAMP`
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [x] I have added an entry to CHANGELOG.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -179,8 +184,8 @@ models:
           interval: 1
 ```
 
-#### at_least_one ([source](macros/schema_tests/at_least_one.sql))
-This schema test asserts if column has at least one value.
+#### at_least_one ([source](macros/generic_tests/at_least_one.sql))
+Asserts if a column has at least one value.
```
Comment (joellabes) [inline README.md:188]:
```suggestion
Asserts that a column has at least one value.
```

File: README.md [hunk 2]
```diff
@@ -194,8 +199,8 @@ models:
           - dbt_utils.at_least_one
 ```
 
-#### not_constant ([source](macros/schema_tests/not_constant.sql))
-This schema test asserts if column does not have same value in all rows.
+#### not_constant ([source](macros/generic_tests/not_constant.sql))
+Asserts if a column does not have the same value in all rows.
```
Comment (joellabes) [inline README.md:203]:
```suggestion
Asserts that a column does not have the same value in all rows.
```

File: README.md [hunk 3]
```diff
@@ -209,8 +214,8 @@ models:
           - dbt_utils.not_constant
 ```
 
-#### cardinality_equality ([source](macros/schema_tests/cardinality_equality.sql))
-This schema test asserts if values in a given column have exactly the same cardinality as values from a different column in a different model.
+#### cardinality_equality ([source](macros/generic_tests/cardinality_equality.sql))
+Asserts if values in a given column have exactly the same cardinality as values from a different column in a different model.
```
Comment (joellabes) [inline README.md:218]:
```suggestion
Asserts that values in a given column have exactly the same cardinality as values from a different column in a different model.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,10 @@
+# dbt-utils v0.8.2
+## Fixes  
+- Fix union_relations error from [#473](https://github.com/dbt-labs/dbt-utils/pull/473) when no include/exclude parameters are provided ([#505](https://github.com/dbt-labs/dbt-utils/issues/505), [#509](https://github.com/dbt-labs/dbt-utils/pull/509))
+
+## Quality of life
+- Updated references to 'schema test' in project file structure and documentation referred to in [#485](https://github.com/dbt-labs/dbt-utils/issues/485)
```
Comment (joellabes) [inline CHANGELOG.md:6]: This should actually be part of 0.8.3 - the changelog should now contains a block for that version if you could move this up please!

**Review by joellabes — CHANGES_REQUESTED** [review]
Hey @b-moynihan, this looks great! Thanks for tidying everything up 🌟 

There's a couple of suggestions attached - mostly swapping _asserts if_ for _asserts that_. I've also changed the base branch to be `next/patch` which brought in a couple of other changes from when you first forked this, but nothing dramatic.

**Comment by joellabes** [comment]
I've also just seen there's some merge conflicts on the changelog and readme - let me know if you need any help teasing them apart 🔀

**Comment by joellabes** [comment]
Note to self when re reviewing this: check that there aren't any configs in dbt_project.yml etc that depend on the current folder structure

File: CHANGELOG.md (hunk 4 above)
Comment (b-moynihan) [reply CHANGELOG.md:6]: Addressed as part of resolving merge conflicts

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
+## New features
+- A macro for deduplicating data ([#335](https://github.com/dbt-labs/dbt-utils/issues/335), [#512](https://github.com/dbt-labs/dbt-utils/pull/512))
 
 ## Quality of life
 - Updated references to 'schema test' in project file structure and documentation referred to in [#485](https://github.com/dbt-labs/dbt-utils/issues/485)
 
-# dbt-utils v0.8.1
+# dbt-utils v0.8.2
+## Fixes
+- Fix union_relations error from [#473](https://github.com/dbt-labs/dbt-utils/pull/473) when no include/exclude parameters are provided ([#505](https://github.com/dbt-labs/dbt-utils/issues/505), [#509](https://github.com/dbt-labs/dbt-utils/pull/509))
```
Comment (joellabes) [inline CHANGELOG.md:11]: Lost the 0.8.1 changelog header
```suggestion
# dbt-utils v0.8.2
## Fixes
- Fix union_relations error from [#473](https://github.com/dbt-labs/dbt-utils/pull/473) when no include/exclude parameters are provided ([#505](https://github.com/dbt-labs/dbt-utils/issues/505), [#509](https://github.com/dbt-labs/dbt-utils/pull/509))

# dbt-utils v0.8.1
```

File: README.md [hunk 6]
```diff
@@ -71,7 +71,7 @@ For compatibility details between versions of dbt-core and dbt-utils, [see this
 ----
 ### Generic Tests
 #### equal_rowcount ([source](macros/generic_tests/equal_rowcount.sql))
-Asserts that two relations have the same number of rows.
+Asserts the that two relations have the same number of rows.
```
Comment (joellabes) [inline README.md:74]:
```suggestion
Asserts that two relations have the same number of rows.
```

**Review by joellabes — APPROVED** [review]
This is looking great! I'll wait for the automated tests to come back successfully and then get this merged in 🎉

**Pull request merged**